### PR TITLE
doc improvement: matter: update mentions of Matter controller and remove "Project CHIP"

### PR DESCRIPTION
**Pull request by greg-fer**
- doc: matter: update mentions of Matter controller
- doc: matter: remove mentions of Project CHIP

### CODE REVIEW

File: doc/nrf/ug_matter_creating_accessory.rst [hunk 1]
```diff
@@ -364,7 +364,7 @@ Testing the new sensor application
 **********************************
 
 .. note::
-   Use the Python Controller when :ref:`setting up Matter development environment <ug_matter_configuring_mobile>`.
+   Use the the mobile Matter controller when :ref:`setting up Matter development environment <ug_matter_configuring_mobile>`.
```
[the remark this answers is not in the document]
Comment (Damian-Nordic) [reply doc/nrf/ug_matter_creating_accessory.rst:367]: Below we need to also update the commands (as those are Python controller commands):
```
./chip-tool onoff on *node_ID* 1
./chip-tool temperaturemeasurement read measured-value *node_ID* 1 
./chip-tool onoff off *node_ID* 1
```

File: doc/nrf/ug_thread_tools.rst [hunk 2]
```diff
@@ -191,7 +191,7 @@ Running OTBR using Docker
 
 For development purposes, you can run the OpenThread Border Router on any Linux-based system using a Docker container that already has the Border Router installed.
 This solution can be used when you are only interested in direct communication between your Border Router and the Thread network.
-For example, you can use the Docker container when you want to establish IP communication between an application running on Linux (such as the :ref:`Python Controller for Matter <ug_matter_configuring>`) and an application running on a Thread node.
+For example, you can use the Docker container when you want to establish IP communication between an application running on Linux (such as the :ref:`CHIP Tool Matter controller <ug_matter_configuring>`) and an application running on a Thread node.
```
Comment (Damian-Nordic) [inline doc/nrf/ug_thread_tools.rst:194]: I think we use a bunch of different names to name the same thing. How about just: "CHIP Tool for Linux or macOS"?

File: samples/matter/template/README.rst [hunk 3]
```diff
@@ -161,7 +161,7 @@ When you have built the sample and programmed it to your development kit, it aut
 At this point, you can press **Button 1** for six seconds to initiate the factory reset of the device.
 
 .. note::
-    If you are new to Matter, commission the Matter device using the Android Mobile Controller when :ref:`setting up the Matter development environment <ug_matter_configuring_mobile>`.
+    If you are new to Matter, commission the Matter device using the Mobile Controller for Android (Android CHIPTool) when :ref:`setting up the Matter development environment <ug_matter_configuring_mobile>`.
```
[the remark this answers is not in the document]
Comment (greg-fer) [reply samples/matter/template/README.rst:164]: The reason for this is that I want to be consistent with the images. :)

File: doc/nrf/ug_matter_configuring_controller.rst [hunk 4]
```diff
@@ -24,59 +24,42 @@ The following figure shows the available Matter controllers in the |NCS|.
...
 
 .. _ug_matter_configuring_controller_mobile:
 
-Mobile controller for Android
-*****************************
+CHIP Tool for Android
+*********************
 
-The Mobile Controller for Android allows you to test Matter applications using an Android smartphone.
+CHIP Tool for Android (also known as Android CHIPTool) is the recommended Mobile controller for mobile, which allows you to test Matter applications using an Android smartphone.
```
Comment (Damian-Nordic) [inline doc/nrf/ug_matter_configuring_controller.rst:56]:
```suggestion
CHIP Tool for Android (also known as Android CHIPTool) is the recommended Matter controller for mobile, which allows you to test Matter applications using an Android smartphone.
```

**Review by Damian-Nordic — APPROVED** [review]

**Review by LuDuda — APPROVED** [review]
Looks great! 

One nit already found by Damian to be fixed.

Also i see two conventions: "CHIP Tool for Android" and "Android CHIPTool" - i know you explained that in the documentation.. but do we need the second type at all?

**Comment by greg-fer** [comment]
@LuDuda , @Damian-Nordic , please just give a one last look, as I replaced more occurrences of Android CHIPTool...

**Review by Damian-Nordic — APPROVED** [review]

**Review by LuDuda — APPROVED** [review]

**Comment by greg-fer** [comment]
@anangl , @shantha-14 , ready for merge, despite Twister. This is doc-only.

**Pull request merged**